### DISCUSSION TITLE
How should call stacks be read?

### DISCUSSION THREAD

**Discussion by nkakouros** [niieani/bash-oo-framework#93] [Q&A]
I have the following stack trace:

![screenshot_20180117_002703](https://user-images.githubusercontent.com/11805218/35017565-aacbb3e2-fb1d-11e7-8bf4-524942d1b66c.png)

This is how I interpret the output. Could you correct me if I am wrong? 
The "important" lines are the ones starting with the arrow (the underlined, red ones). I call them main lines. The first of these, the top-most, contains a friendly error message. The others show at most 3 things:
- the red part is the function in whose call the error happened
- the white part are the arguments passed to that function
- the blue part which is the file and line of the function call

Then, some times there might be a second accompanying line for these main lines. These accompanying lines are present when the main line does not provide the full picture, ie the actual function/builtin call with the arguments passed. 

Is the above correct? 

Some more questions: 

1. What does the 'X' icon in the accompanying lines mean?
2. In the example picture, the second call from the top is to the `Variable::TrapAssignNumberedParameter` function. But the accompanying line seems out of place. Is this due to the magic with the named parameters aliases and traps?

**Comment by niieani** [comment]
Hey @tterranigma, I'll have to get back to you on this one since I'm rather busy lately.
Cheers
